src/Microsoft.AspNet.Diagnostics/Views/BaseView.cs

```diff
@@ -117,6 +118,48 @@ protected void WriteLiteral(object value)
                 values);
         }
 
+        private List<string> _attributeValues { get; set; }
+
+        protected void WriteAttributeValue(string thingy, int startPostion, object value, int endValue, int dealyo, bool yesno)
+        {
+            if(_attributeValues == null)
+            {
+                _attributeValues = new List<string>();
+            }
+
+            _attributeValues.Add(value.ToString());
+        }
+
+        private string _attributeEnding { get; set; }
+
+        protected void BeginWriteAttribute(string name, string begining, int startPosition, string ending, int endPosition, int thingy)
+        {
+            if(!string.IsNullOrEmpty(_attributeEnding))
+            {
+                throw new InvalidOperationException();
+            }
+
+            Output.Write(begining);
+            _attributeEnding = ending;
+        }
+
+        protected void EndWriteAttribute()
+        {
+            if(string.IsNullOrEmpty(_attributeEnding))
+            {
+                throw new InvalidOperationException();
```

Comment:
I'm guessing you mean to replace that block with Debug.Assert(string.IsNullOrEmpty(AttributeEnding), "AttributeEnding can't be null or empty") ?